app/App/Main/Account/Requests/TransactionRequest/AdjustFee/index.js

```diff
@@ -50,329 +209,44 @@ class TxFeeOverlay extends React.Component {
-                if (isNaN(gasPrice)) return
-                gasPrice = this.trimGwei(this.limitRange(gasPrice + 1, 0, 9999))
-                gasPrice = this.toDisplayFromGwei(BigNumber(gasPrice))
-                this.setGasPrice(gasPrice)
-              } else if (e.key === 'ArrowDown') {
-                e.preventDefault()
-                let gasPrice = parseFloat(this.state.gasPrice)
-                if (isNaN(gasPrice)) return
-                gasPrice = this.trimGwei(this.limitRange(gasPrice - 1, 0, 9999))
-                gasPrice = this.toDisplayFromGwei(BigNumber(gasPrice))
-                this.setGasPrice(gasPrice)
-              }
-            }}
-          />
-        </div>
-        <div className='txFeeOverlayLabel'>Gas Price (GWEI)</div>
-      </div>
-    )
-  }
+  render () {
+    const { req: { data, handlerId } } = this.props
+    const maxTotalFee = maxFee(data)
 
-  renderGasLimitInput () {
     return (
-      <div className='txFeeOverlayLimit'>
-        <div className='txFeeOverlayInput'>
-          <input 
-            tabIndex={3} 
-            value={this.state.gasLimit}
-            className='txFeeOverlayInput' 
-            onChange={(e) => {
-              this.setGasLimit(e.target.value.match('[0-9]*'))
-            }}
-            onKeyDown={(e) => {
-              if (e.key === 'Enter') {
-                e.preventDefault()
-                e.target.blur()
-              } else if (e.key === 'ArrowUp') {
-                e.preventDefault()
-                const gasLimit = parseInt(this.state.gasLimit) + 1000
-                if (isNaN(gasLimit)) return
-                this.setGasLimit(gasLimit.toString())
-              } else if (e.key === 'ArrowDown') {
-                e.preventDefault()
-                const gasLimit = parseInt(this.state.gasLimit) - 1000
-                if (isNaN(gasLimit)) return
-                this.setGasLimit(gasLimit.toString())
-              }
-            }}
-          />
-        </div>
-        <div className='txFeeOverlayLabel'>Gas Limit (UNITS)</div>
+      <div className='txAdjustFee cardShow' ref={this.moduleRef}>
+        {
+          usesBaseFee(data) ? 
+          <>
+            <BaseFeeInput initialValue={this.state.baseFee} maxTotalFee={maxTotalFee} priorityFee={this.state.priorityFee} gasLimit={this.state.gasLimit} handlerId={handlerId} />
+            <PriorityFeeInput initialValue={this.state.priorityFee} maxTotalFee={maxTotalFee} baseFee={this.state.baseFee} gasLimit={this.state.gasLimit} handlerId={handlerId} />
+          </> :  
+          <GasPriceInput initialValue={this.state.gasPrice} maxTotalFee={maxTotalFee} gasLimit={this.state.gasLimit} handlerId={handlerId} />
+        }
+        
+        <GasLimitInput initialValue={this.state.gasLimit} maxTotalFee={maxTotalFee} baseFee={this.state.baseFee} priorityFee={this.state.priorityFee} gasPrice={this.state.gasPrice} handlerId={handlerId} limiter={(val) => limitRange(val, 0, 12.5e6)} />
```

Comment:
does this state ever update? if not I don't think this will work as the other components need the updated gas limit to recalculate their values

Reply:
Yeah I don't think it will update, I had this as a mental TODO.  Going to do tests first

Reply:
since all of these components need access to the state (ie all 3 values are based on all the inputs) we may have to share the state in the top level component and update it using functions we pass into the individual input components

Reply:
looking at this again I don't think the state updates - reason being that if a gas price updates whilst you're editing it that would make the form basically unusable.

Reply:
As discussed we might look to simplify things here by giving the top level component a submit button...